### PR TITLE
Jasssonpet/returns-retained

### PR DESCRIPTION
**Pull request by jasssonpet**

### CODE REVIEW

File: src/Meta/MetaFactory.cpp [hunk 1]
```diff
@@ -176,13 +182,28 @@ void MetaFactory::createFromFunction(const clang::FunctionDecl& function, Functi
...
+
+    bool returnsRetained = Utils::getAttributes<clang::NSReturnsRetainedAttr>(function).size() > 0 || Utils::getAttributes<clang::CFReturnsRetainedAttr>(function).size() > 0;
+    bool returnsNotRetained = Utils::getAttributes<clang::NSReturnsNotRetainedAttr>(function).size() > 0 || Utils::getAttributes<clang::CFReturnsNotRetainedAttr>(function).size() > 0;
+
+    // Clang doesn't handle The Create Rule automatically like for methods, so we have to do it manually
+    if (!(returnsRetained || returnsNotRetained)) {
+        Type* returnType = functionMeta.signature[0];
+        if (returnType->is(TypeInterface) || returnType->is(TypeBridgedInterface) || returnType->is(TypeId)) {
+            std::string functionName = function.getNameAsString();
+            if (functionName.find("Create") != string::npos || functionName.find("Copy") != string::npos) {
```
Comment (jasssonpet) [inline src/Meta/MetaFactory.cpp:200]: Swift is also handling names like `createObject` and `object_create`, should we try to be more clever here?

Comment (fealebenpae) [inline src/Meta/MetaFactory.cpp:200]: Is there an API in the iOS SDK with a name like that? I'd rather not try to second-guess Swift too much, at least not until it's open-sourced so we can see what's going on. If the CF part of Swift even gets open-sourced.

Comment (jasssonpet) [inline src/Meta/MetaFactory.cpp:200]: No, there isn't such in the iOS SDK (except `dispatch_data_create` and the like).

Comment (fealebenpae) [inline src/Meta/MetaFactory.cpp:200]: I suggest ignoring those extra cases, for the time being.

**Pull request merged**